src/matter/cluster/server/GeneralCommissioningServer.ts

```diff
@@ -14,9 +18,18 @@ export const GeneralCommissioningClusterHandler: ClusterServerHandlers<typeof Ge
         return SuccessResponse;
     },
 
-    updateRegulatoryConfig: async ({request: {breadcrumbStep, config}, attributes: {breadcrumb, regulatoryConfig}}) => {
+    setRegulatoryConfig: async ({request: {breadcrumbStep, newRegulatoryConfig, countryCode}, attributes: {breadcrumb, regulatoryConfig, locationCapability}}) => {
+        const locationCapabilityValue = locationCapability.get();
+        if (locationCapabilityValue !== RegulatoryLocationType.IndoorOutdoor && newRegulatoryConfig !== locationCapabilityValue) {
+            return {errorCode: CommissioningError.ValueOutsideRange, debugText: "Invalid regulatory location"};
+        }
+        if (locationCapabilityValue === RegulatoryLocationType.IndoorOutdoor) {
```

Comment:
This if is not necessary since this is done internally by the attribute and might lead to issue if more RegulatoryLocationType are introduced in the future.